src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/QueueCapacity.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions;
+
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.CPU;
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.HEAP;
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension.NETWORK;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.ThreadPoolEnum;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class QueueCapacity implements Action {
+
+  public static final String NAME = "queue_capacity";
+  public static final int COOL_OFF_PERIOD_IN_SECONDS = 300;
```

Comment:
Do we want to make it configurable ?

Reply:
This timeout is independent of its calling function, so we shouldn't need it configurable via callers. Did you mean reading and setting this value from rca.conf? I'm not sure if we'll want to edit this at runtime (maybe we can make that change when we add rca.conf integration to this class, or if we see a need later).

Reply:
Ya, I meant it reading from rca.conf. If you want to do an A/B test to come up with optimal value, keeping it configurable might help. Otherwise, we will have to deploy jars with the changes.

Reply:
Let's discuss and track this separately. If needed, we can add this along with the other rca.conf integrations which are pending.